integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datacompaction/CarbonIndexFileMergeTestCase.scala

```diff
@@ -300,8 +300,8 @@ class CarbonIndexFileMergeTestCase
     assert(getIndexFileCount("default_nonindexmerge", "1") == 100)
     assert(getIndexFileCount("default_nonindexmerge", "2") == 100)
     assert(getIndexFileCount("default_nonindexmerge", "3") == 100)
-    assert(getIndexFileCount("default_nonindexmerge", "0.1") == 100)
```

Comment:
Why test is changed?

Reply:
previously merge index is triggered at the end of load. So validation for valid segments was done. as 0.1 and 2.1 are invalid segments(as they are merged to 0.2), they dont have mergeindex files.
Current implementation will trigger merge index for all the segments written, in load or compaction case. So 0.1 and 2,1 also have merge index